src/rendervulkan.cpp

```diff
@@ -283,62 +313,72 @@ bool CVulkanTexture::BInit( uint32_t width, uint32_t height, VkFormat format, cr
 	imageInfo.usage = usage;
 	imageInfo.samples = VK_SAMPLE_COUNT_1_BIT;
 	imageInfo.sharingMode = VK_SHARING_MODE_EXCLUSIVE;
-	
+
 	if ( pDMA != nullptr )
 	{
 		assert( format == DRMFormatToVulkan( pDMA->format ) );
 	}
-	
+
 	if ( g_vulkanSupportsModifiers && pDMA && pDMA->modifier != DRM_FORMAT_MOD_INVALID )
 	{
-		VkPhysicalDeviceImageDrmFormatModifierInfoEXT modifierFormatInfo = {};
-		VkPhysicalDeviceExternalImageFormatInfo externalImageFormatInfo = {};
-		VkPhysicalDeviceImageFormatInfo2 imageFormatInfo = {};
 		VkExternalImageFormatProperties externalImageProperties = {};
-		VkImageFormatProperties2 imageProperties = {};
-		
-		modifierFormatInfo.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_IMAGE_DRM_FORMAT_MODIFIER_INFO_EXT;
-		modifierFormatInfo.drmFormatModifier = pDMA->modifier;
-		modifierFormatInfo.sharingMode = imageInfo.sharingMode;
-
-		externalImageFormatInfo.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_EXTERNAL_IMAGE_FORMAT_INFO;
-		externalImageFormatInfo.pNext = &modifierFormatInfo;
-		externalImageFormatInfo.handleType = VK_EXTERNAL_MEMORY_HANDLE_TYPE_DMA_BUF_BIT_EXT;
-
-		imageFormatInfo.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_IMAGE_FORMAT_INFO_2;
-		imageFormatInfo.pNext = &externalImageFormatInfo;
-		imageFormatInfo.format = imageInfo.format;
-		imageFormatInfo.type = imageInfo.imageType;
-		imageFormatInfo.tiling = VK_IMAGE_TILING_DRM_FORMAT_MODIFIER_EXT;
-		imageFormatInfo.usage = imageInfo.usage;
-		imageFormatInfo.flags = imageInfo.flags;
-		
 		externalImageProperties.sType = VK_STRUCTURE_TYPE_EXTERNAL_IMAGE_FORMAT_PROPERTIES;
-
-		imageProperties.sType = VK_STRUCTURE_TYPE_IMAGE_FORMAT_PROPERTIES_2;
-		imageProperties.pNext = &externalImageProperties;
-		
-		if ( vkGetPhysicalDeviceImageFormatProperties2(physicalDevice, &imageFormatInfo, &imageProperties) == VK_SUCCESS &&
+		if ( getModifierProps( &imageInfo, pDMA->modifier, &externalImageProperties ) == VK_SUCCESS &&
 		     ( externalImageProperties.externalMemoryProperties.externalMemoryFeatures & VK_EXTERNAL_MEMORY_FEATURE_IMPORTABLE_BIT ) )
 		{
 			modifierInfo.sType = VK_STRUCTURE_TYPE_IMAGE_DRM_FORMAT_MODIFIER_EXPLICIT_CREATE_INFO_EXT;
 			modifierInfo.pNext = imageInfo.pNext;
 			modifierInfo.drmFormatModifier = pDMA->modifier;
 			modifierInfo.drmFormatModifierPlaneCount = pDMA->n_planes;
 			modifierInfo.pPlaneLayouts = modifierPlaneLayouts;
-		
+
 			for ( int i = 0; i < pDMA->n_planes; ++i )
 			{
 				modifierPlaneLayouts[i].offset = pDMA->offset[i];
 				modifierPlaneLayouts[i].rowPitch = pDMA->stride[i];
 			}
-		
+
 			imageInfo.pNext = &modifierInfo;
-		
+
 			imageInfo.tiling = tiling = VK_IMAGE_TILING_DRM_FORMAT_MODIFIER_EXT;
 		}
 	}
-	
+
+	std::vector<uint64_t> modifiers = {};
```

Comment:
Move modifiers into the if?

Reply:
Unfortunately that means it'll get free'd at the end of the if block, prior to the `vkCreateImage` call.

Reply:
Oops, right